src/app/util/util.cpp

```diff
@@ -515,6 +515,11 @@ void emberAfCopyLongString(uint8_t * dest, const uint8_t * src, size_t size)
 // default value of NULL is treated as all zeroes.
 int8_t emberAfCompareValues(const uint8_t * val1, const uint8_t * val2, uint16_t len, bool signedNumber)
 {
+    if (len == 0)
```

Comment:
I'm curious: why is this test needed here?  What code below tickes clang-tidy if len == 0?

Reply:
Ah, it's the `1 << (8* len - 1)` bits below.  OK.